PyDSS/pyPostprocessor/PostprocessScripts/AutomatedVoltageUpgrade.py

```diff
@@ -1,1855 +1,2318 @@
+                            self.cluster_optimal_reg_nodes["sub_LTC"] = [self.severity_indices[2]]
+                            self.sub_LTC_added_flag = 1
+                            if len(self.buses_with_violations) > 0:
+                                self.LTC_controls_sweep(upper_limit=self.upper_limit, lower_limit=self.lower_limit)
+                                self.check_voltage_violations_multi_tps(upper_limit=self.upper_limit,
+                                                                        lower_limit=self.lower_limit)
+                                if self.config["create_topology_plots"]:
+                                    self.plot_violations()
+                    elif dss.RegControls.Count() == 0 and len(self.buses_with_violations) > 0:
+                        self.add_substation_LTC()
+                        self.check_voltage_violations_multi_tps(upper_limit=self.upper_limit,
+                                                                lower_limit=self.lower_limit)
+                        self.cluster_optimal_reg_nodes["sub_LTC"] = [self.severity_indices[2]]
+                        self.sub_LTC_added_flag = 1
+                        if len(self.buses_with_violations) > 0:
+                            self.LTC_controls_sweep(upper_limit=self.upper_limit, lower_limit=self.lower_limit)
+                            self.check_voltage_violations_multi_tps(upper_limit=self.upper_limit,
+                                                                    lower_limit=self.lower_limit)
+                            if self.config["create_topology_plots"]:
+                                self.plot_violations()
+
+            # Correct regulator settings if regs are present in the feeder other than the sub station LTC
+            # TODO: Remove regs of last iteration
+            self.check_voltage_violations_multi_tps(upper_limit=self.upper_limit,
+                                                    lower_limit=self.lower_limit)
+
+            self.logger.info(f"Total number of buses in circuit: {len(dss.Circuit.AllBusNames())}")
+            # if number of buses with violations is very high, the loop for adding new regulators will take very long
+            # so disable this block - current condition is if number of violations > 250
+            if len(self.buses_with_violations) >= min((100 * len(self.initial_buses_with_violations)), 500,
+                                                     len(dss.Circuit.AllBusNames())
+                                                     ):
+                self.logger.info(f"At this point, number of buses with violations is {len(self.buses_with_violations)},"
+                                 f" but initial number of buses with violations is "
+                                 f"{len(self.initial_buses_with_violations)}")
+                self.logger.info("So disable option for addition of new regulators")
+                self.place_new_regulators = False

```

Comment:
Should this be False?

Reply:
Checked. There is another flag above that sets self.place_new_regulators = True (which is what we want by default)

In this condition, this flag is set to False if the number of violations is too large.

So this is correct.